typescript/src/schema/proto.ts

```diff
@@ -586,12 +586,16 @@ function augmentMethod(
 
 export function getSingleHeaderParam(
   rule: protos.google.api.IHttpRule
-): string[] {
+): string[][] {
   const message =
     rule.post || rule.delete || rule.get || rule.put || rule.patch;
   if (message) {
-    const res = message.match(/{(.*?)[=}]/);
-    return res?.[1] ? res[1].split('.') : [];
+    const result: string[][] = [];
+    const matches = message.matchAll(/{(.*?)[=}]/g);
+    for (const match of matches) {
+      result.push(match[1] ? match[1].split('.') : []);
```

Comment:
Why do we need to push an empty array here? Just push `if (match[1]) { result.push(...); }`

Reply:
Good point. Will update on this.